src/Adapter/Order/CommandHandler/SendProcessOrderEmailHandler.php

```diff
@@ -0,0 +1,198 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Order\CommandHandler;
+
+use Cart;
+use Customer;
+use Language;
+use Link;
+use Mail;
+use PrestaShop\PrestaShop\Core\Domain\Cart\Exception\CartNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Cart\ValueObject\CartId;
+use PrestaShop\PrestaShop\Core\Domain\Customer\Exception\CustomerNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Customer\ValueObject\CustomerId;
+use PrestaShop\PrestaShop\Core\Domain\Order\Command\SendProcessOrderEmailCommand;
+use PrestaShop\PrestaShop\Core\Domain\Order\CommandHandler\SendProcessOrderEmailHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Order\Exception\OrderEmailSendException;
+use PrestaShop\PrestaShop\Core\Domain\Order\Exception\OrderException;
+use PrestaShopException;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Handles SendProcessOrderEmail command using legacy object model
+ */
+class SendProcessOrderEmailHandler implements SendProcessOrderEmailHandlerInterface
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var Link
+     */
+    private $contextLink;
+
+    public function __construct(
+        TranslatorInterface $translator,
+        Link $contextLink
+    ) {
+        $this->translator = $translator;
+        $this->contextLink = $contextLink;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(SendProcessOrderEmailCommand $command): void
+    {
+        $cartId = $command->getCartId();
+
+        try {
+            $cart = $this->getCart($cartId);
+            $customer = $this->getCustomer(new CustomerId((int) $cart->id_customer));
```

Comment:
Indeed it's used for validation purpose but I think it's valid.
The reason why it was not provided as CustomerId straight away is that data comes from legacy object Cart. If the code was fully migrated, it would be a domain Cart Object with type-hinted CustomerId. So the type-hinting is done a bit late but is still relevant.

Reply:
Mmm, it doesn't matter if it comes from a legacy or something else, if we want to use the ValueObject either it is used as a parameter of the commande or the command can have an `int` parameter but store it in the value object This allows to perform the check earlier, there's no interest in creating it later in the handler